service/frontend/adminHandler.go

```diff
@@ -948,42 +948,52 @@ func (adh *AdminHandler) GetWorkflowExecutionRawHistoryV2(ctx context.Context, r
 		execution.GetWorkflowId(),
 		adh.numberOfHistoryShards,
 	)
-	rawHistoryResponse, err := adh.persistenceExecutionManager.ReadRawHistoryBranch(ctx, &persistence.ReadHistoryBranchRequest{
-		BranchToken: targetVersionHistory.GetBranchToken(),
-		// GetWorkflowExecutionRawHistoryV2 is exclusive exclusive.
-		// ReadRawHistoryBranch is inclusive exclusive.
-		MinEventID:    pageToken.GetStartEventId() + 1,
-		MaxEventID:    pageToken.GetEndEventId(),
-		PageSize:      pageSize,
-		NextPageToken: pageToken.PersistenceToken,
-		ShardID:       shardID,
-	})
-	if err != nil {
-		if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
-			// when no events can be returned from DB, DB layer will return
-			// EntityNotExistsError, this API shall return empty response
-			return &adminservice.GetWorkflowExecutionRawHistoryV2Response{
-				HistoryBatches: []*commonpb.DataBlob{},
-				NextPageToken:  nil, // no further pagination
-				VersionHistory: targetVersionHistory,
-			}, nil
+
+	var historyEventBlobs []*commonpb.DataBlob
+	var nodeIDs []int64
+	var size int64
+	if adh.config.readEventsFromHistory() {
+		panic("not implemented yet")
+	} else {
+		rawHistoryResponse, err := adh.persistenceExecutionManager.ReadRawHistoryBranch(ctx, &persistence.ReadHistoryBranchRequest{
+			BranchToken: targetVersionHistory.GetBranchToken(),
+			// GetWorkflowExecutionRawHistoryV2 is exclusive exclusive.
+			// ReadRawHistoryBranch is inclusive exclusive.
+			MinEventID:    pageToken.GetStartEventId() + 1,
+			MaxEventID:    pageToken.GetEndEventId(),
+			PageSize:      pageSize,
+			NextPageToken: pageToken.PersistenceToken,
+			ShardID:       shardID,
+		})
+		if err != nil {
+			if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
+				// when no events can be returned from DB, DB layer will return
+				// EntityNotExistsError, this API shall return empty response
+				return &adminservice.GetWorkflowExecutionRawHistoryV2Response{
+					HistoryBatches: []*commonpb.DataBlob{},
+					NextPageToken:  nil, // no further pagination
+					VersionHistory: targetVersionHistory,
+				}, nil
+			}
+			return nil, err
 		}
```

Comment:
I don't think this should be committed--what happens if someone sets the dc to 100% in a later release and then rolls back to this one? Or, what if someone sets the DC to a non-zero number now? We should add the implementation first, then the DC branching